scripts/test-nix-cache.sh

```diff
@@ -0,0 +1,217 @@
+#!/usr/bin/env bash
+# shellcheck disable=SC2181,SC2059
+
+# SPDX-FileCopyrightText: 2022-2024 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+
+set -e # exit immediately if a command fails
+set -u # treat unset variables as an error and exit
+set -o pipefail # exit if any pipeline command fails
+
+# This script is a helper to benchmark nix cache download
+
+################################################################################
+
+MYNAME=$(basename "$0")
+TMPSTOREDIR="$(mktemp -d --suffix .nix.store)"
+RED='' GREEN='' NONE=''
+
+################################################################################
+
+usage () {
+    echo "Usage: $MYNAME [-h] [-v] [-o OUTFILE] -b BINCACHE -k KEY -f FLAKEREF "
+    echo ""
+    echo "Perform basic download test for the given binary cache."
+    echo "The script attempts to download the FLAKEREF closure from the given"
```

Comment:
I'd just have it ask for a store path to download, instead of a flakeref.

This allows moving the evaluation outside of this script - we might already have a store path calculated elsewhere.

Reply:
No, you can only point it to a nix file. I was essentially suggesting to move the nix invocation in L175 out of this script to somewhere else, and pass it in.

However, on a second look, I see you not only use this to evaluate, but also to substitute through Nix. So it's probably fine as-is.

Reply:
What I meant was both below commands should work, and the later one skips evaluation:

```
❯ sudo ./scripts/test-nix-cache.sh\
     -f nixpkgs#firefox\
     -b https://cache.nixos.org\
     -k 'cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY='

Testing 'https://cache.nixos.org':
  test_nix_cache_info ...                                   SUCCESS (0 sec)
  test_substitute_flakeref ...                              SUCCESS (29 sec)
```

```
❯ sudo ./scripts/test-nix-cache.sh\
     -f /nix/store/ka2232kk5m6z6i9rdbhw5pbvyhhbja77-firefox-129.0\
     -b https://cache.nixos.org\
     -k 'cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY='

Testing 'https://cache.nixos.org':
  test_nix_cache_info ...                                   SUCCESS (1 sec)
  test_substitute_flakeref ...                              SUCCESS (24 sec)
```